src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/SymbolMatcherTests.cs

```diff
@@ -1404,5 +1404,151 @@ static void verify(string source1, string source2)
                 Assert.Null(matcher.MapDefinition(f_1.GetCciAdapter()));
             }
         }
+
+        [Fact]
+        public void Record_ImplementSynthesizedMember_ToString()
+        {
+            var source0 = @"
+public record R
+{
+}";
+            var source1 = @"
+public record R
+{
+    public override string ToString() => ""R"";
+}";
+            var compilation0 = CreateCompilation(source0, options: TestOptions.DebugDll);
+            var compilation1 = compilation0.WithSource(source1);
+
+            var matcher = new CSharpSymbolMatcher(
+                null,
+                compilation1.SourceAssembly,
+                default,
+                compilation0.SourceAssembly,
+                default,
+                null);
+
+            var member = compilation1.GetMember<SourceOrdinaryMethodSymbol>("R.ToString");
+            var other = matcher.MapDefinition(member.GetCciAdapter());
+            Assert.NotNull(other);
+        }
+
+        [Fact]
+        public void Record_ImplementSynthesizedMember_PrintMembers()
+        {
+            var source0 = @"
+public record R
+{
+}";
+            var source1 = @"
+public record R
+{
+    protected virtual bool PrintMembers(System.Text.StringBuilder builder) => true;
+}";
+            var compilation0 = CreateCompilation(source0, options: TestOptions.DebugDll);
+            var compilation1 = compilation0.WithSource(source1);
+
+            var matcher = new CSharpSymbolMatcher(
+                null,
+                compilation1.SourceAssembly,
+                default,
+                compilation0.SourceAssembly,
+                default,
+                null);
+
+            var member = compilation1.GetMember<SourceOrdinaryMethodSymbol>("R.PrintMembers");
```

Comment:
Still working on this, it needs expanding to map from this symbol to the compiler generated one in compilation0, but that fails at the moment.